src/shared/shared-types.d.ts

```diff
@@ -129,7 +129,17 @@ export type JsonLocations = {
   marker: string
 }[] // ReturnType<typeof DeltaChat.prototype.getLocations>
 
-export type JsonMessage = ReturnType<typeof Message.prototype.toJson>
+export type JsonMessage = ReturnType<typeof Message.prototype.toJson> & {
+  quote: MessageQuote | null
+}
+
+export type MessageQuote = {
+  messageId: number
+  text: string
+  displayName: string
+  displayColor: string
+  overrideSenderName: string
+}
```

Comment:
this type needs to have variant without `displayName`, `displayColor` and `overrideSenderName` for the case that the quoted message is not found in the database (for example when that message was deleted.

Reply:
And also messageId i guess?

Reply:
yes, see https://js.delta.chat/classes/message.html#getquotedmessage message can be null.